meta/language.LL1

```diff
@@ -0,0 +1,178 @@
+                 | [Expression] /!?\./ identifier
+                 | Expression "(" ParameterList ")"
+
+Statement        = Expression  [";"]
+                 | DeclarationStatement
+                 | AssignmentStatement
+                 | ReturnStatement
+                 | YieldStatement
+                 | BlockStatement
+                 | ThrowStatement
+                 | IfStatement
+                 | TryStatement
+                 | ForStatement
+                 | LoopStatement
+                 | WhileStatement
+                 | "defer" Statement
+                 | "unsafe" BlockStatement
+                 | "cpp" "{" {String} "}"
+                 | "continue"
+                 | "break"
+
+DeclarationStatement = /let|mut/ identifier [":" TypeExpression] "=" Expression [";"]
+AssignmentStatement  = identifier "=" Expression [";"]
+ReturnStatement      = "return" [Expression] [";"]
+YieldStatement       = "yield" [Expression] [";"]
+ThrowStatement       = "throw" [Expression] [";"]
+
+BlockStatement       = "{" Statement "}"
+
+IfStatement          = "if" BooleanExpression BlockStatement ["else" BlockStatement]
+TryStatement         = "try" (Expression | BlockStatement) "catch" identifier BlockStatement
+ForStatement         = "for" identifier "in" Expression BlockStatement
+LoopStatement        = "loop" BlockStatement
+WhileStatement       = "while" Expression BlockStatement
+
+// NOTE: `mut this` is only allowed in ClassFunctions
+ArgumentList   = [[["mut"] "this"] [Arguments]]
+Arguments      = "," Argument [Arguments]
+Argument       = ["anon"] ["mut"] identifier [":" TypeExpression]
+
+ParameterList  = [Parameter [Parameters]]
+Parameters     = "," Parameter [Parameters]
+Parameter      = [identifier ":"] Expression
+
+Declaration    = Comment
+               | Namespace
+               | Import
+               | Function
+               | Class
+               | Enum
+
+Comment        = /\/\/.*\n/
+Namespace      = "namespace" identifier "{" {Declaration} "}"
+Import         = "import" identifier ["{" identifier {"," identifier} "}"]
+Function       = ["extern"] "function" identifier [GenericSpecifier] "(" ArgumentList ")" ["->" TypeExpression] ["throws"] [("{" {Statement} "}") | ("=>" Expression)]
+
+// NOTE: This describes Classes and Structs
+Class          = ["extern"] ("class"|"struct") identifier "{" {ClassMembers} "}"
+ClassMembers   = ClassFunction | ClassMember
+ClassMember    = [AccessModifier ["mut"]] identifier ":" TypeExpression
+ClassFunction  = [AccessModifier] "function" identifier [GenericSpecifier] "(" ArgumentList ")" ["->" TypeExpression] ["throws"] (("{" {Statement} "}") | ("=>" Expression))
+AccessModifier = "public"
+               | "private"
+               | "restricted" "(" identifier [GenericSpecifier] {"," identifier [GenericSpecifier]} ")"
+
+Enum           = ["boxed"] "enum" identifier [GenericSpecifier] [":" TypeExpression] "{" EnumMember {EnumMember} "}"
```

Comment:
`GenericSpecifier` and `":" TypeExpression` are mutually exclusive (and so is `"boxed"` with `":" TypeExpression`).